### ISSUE TITLE
knife bootstrap through ssh fails

### ISSUE THREAD

**Issue by anuriq** [chef/knife-windows#372]
I am using FreeSSHd on windows nodes. Everything worked fine until newest release of Chef DK.

Error:
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/knife-windows-1.4.1/lib/chef/knife/bootstrap_windows_base.rb:429:in `set_target_architecture': undefined method `relay_winrm_command' for #<Chef::Knife::BootstrapWindowsSsh:0x007fb8cc319de0> (NoMethodError)

Full debug log available [here](https://gist.github.com/anuriq/04bf5d6d5eb3749a68e83c41a5edfd68)

**Comment by anuriq** [comment]
btw bootstrap through winrm works fine.

**Comment by peterdeme** [comment]
Hi anuriq

I faced the same issue. Not sure what relay_winrm_command do but instead I manually modified the os_archtiectures like this:

```
 def set_target_architecture(bootstrap_architecture)
     
        current_architecture = : :x86_64

        if bootstrap_architecture.nil?
          architecture = current_architecture
        elsif bootstrap_architecture == :x86_64 && current_architecture == :i386
          raise "You specified bootstrap_architecture as x86_64 but the target machine is i386. A 64 bit program cannot run on a 32 bit machine."
        else
          architecture = bootstrap_architecture
        end

        # The windows install script wants i686, not i386
        architecture = :i686 if architecture == :i386
        Chef::Config[:knife][:architecture] = architecture
```

you have to use sudo to modify that ruby file.

**Comment by mwrock** [comment]
ugh. this is definitely a new bug that breaks ssh based bootstrap on windows.

**Comment by anuriq** [comment]
A call to `relay_winrm_command` should not be in `bootstrap_windows_base.rb`. Maybe it was meant to be in `bootstrap_windows_winrm.rb`. I'd sent a PR if I knew what was the intention...

**Comment by mwrock** [comment]
I think it was just added here not thinking of ssh. The fact is that few use this gem using ssh and so it often is forgotten about. We should really have some integration tests to protect against this. Anyways yeah I think the right approach would be to abstract away the transport and have `bootstrap_windows_base.rb` call that abstraction instead of a specific transport.

**Comment by mwrock** [comment]
fixed via #380
